navjson/navjson.go

```diff
@@ -49,6 +51,31 @@ func Parse(data string) (NavJSON, error) {
 	return j, err
 }
 
+// EnsureFilesExist returns an error if any referenced files do not exist or are directories.
+func EnsureFilesExist(dir string, files []NavFile) error {
+	for _, f := range files {
+		p := path.Join(dir, f.Path, f.Name)
```

Comment:
The file to check if it exists depends on the accumulation of `name` attribute values all the way from the top level parent of the `files` attribute, and not by concatenating f.Path, f.Name.

Also, the existence of files is something that should be done after the `docs` are actually generated, meaning after the markdown file are transformed into html files.

For example, if we take the `attest-docs` repo:

The npm script which converts all md to html is - https://github.com/dequelabs/attest-docs/blob/master/package.json#L10. The ensure files exist should run after the above step against the `dist-docs` directory.

And the path of the file to verify for this leaf in `nav.json` - https://github.com/dequelabs/attest-docs/blob/master/docs/nav.json#L55, would be `/getting-started/npm-setup/generate-api-key.html`.

Reply:
Let's pair on this together and figure out what we can do.